tests/bluetooth/bsim_bt/bsim_test_eatt_chann_sel/src/client_test.c

```diff
@@ -0,0 +1,273 @@
+		if (err != 0) {
+			FAIL("Discover failed (err %d)\n", err);
+		}
+
+		return BT_GATT_ITER_STOP;
+	} else if (params->type == BT_GATT_DISCOVER_CHARACTERISTIC) {
+		const struct bt_gatt_chrc *chrc = (struct bt_gatt_chrc *)attr->user_data;
+
+		if (bt_uuid_cmp(chrc->uuid, TEST_CHRC_UUID) == 0) {
+			printk("Found chrc\n");
+			chrc_handle = chrc->value_handle;
+		}
+	}
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static void gatt_discover(void)
+{
+	static struct bt_gatt_discover_params discover_params;
+	int err;
+
+	printk("Discovering services and characteristics\n");
+
+	discover_params.uuid = test_svc_uuid;
+	discover_params.func = discover_func;
+	discover_params.start_handle = BT_ATT_FIRST_ATTRIBUTE_HANDLE;
+	discover_params.end_handle = BT_ATT_LAST_ATTRIBUTE_HANDLE;
+	discover_params.type = BT_GATT_DISCOVER_PRIMARY;
+
+	err = bt_gatt_discover(g_conn, &discover_params);
+	if (err != 0) {
+		FAIL("Discover failed(err %d)\n", err);
+	}
+
+	WAIT_FOR_FLAG(flag_discover_complete);
+	printk("Discover complete\n");
+}
+
+static void test_subscribed(struct bt_conn *conn, uint8_t err, struct bt_gatt_write_params *params)
+{
+	if (err) {
+		FAIL("Subscribe failed (err %d)\n", err);
+	}
+
+	SET_FLAG(flag_subscribed);
+
+	if (!params) {
+		printk("params NULL\n");
+		return;
+	}
+
+	if (params->handle == chrc_handle) {
+		printk("Subscribed to characteristic\n");
+	} else {
+		FAIL("Unknown handle %d\n", params->handle);
+	}
+}
+
+static volatile size_t num_notifications;
+uint8_t test_notify(struct bt_conn *conn, struct bt_gatt_subscribe_params *params, const void *data,
+		    uint16_t length)
+{
+	printk("Received notification #%u with length %d\n", num_notifications++, length);
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static struct bt_gatt_discover_params disc_params;
+static struct bt_gatt_subscribe_params sub_params = {
+	.notify = test_notify,
+	.write = test_subscribed,
+	.ccc_handle = 0, /* Auto-discover CCC*/
+	.disc_params = &disc_params, /* Auto-discover CCC */
+	.end_handle = BT_ATT_LAST_ATTRIBUTE_HANDLE,
+	.value = BT_GATT_CCC_NOTIFY,
+};
+
+static void gatt_subscribe(void)
+{
+	int err;
+
+	sub_params.value_handle = chrc_handle;
+	err = bt_gatt_subscribe(g_conn, &sub_params);
+	if (err < 0) {
+		FAIL("Failed to subscribe\n");
+	} else {
+		printk("Subscribe request sent\n");
+	}
+	WAIT_FOR_FLAG(flag_subscribed);
+}
+
+static void test_main(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+	if (err != 0) {
+		FAIL("Bluetooth discover failed (err %d)\n", err);
+	}
+
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, device_found);
+	if (err != 0) {
+		FAIL("Scanning failed to start (err %d)\n", err);
+	}
+
+	printk("Scanning successfully started\n");
+
+	WAIT_FOR_FLAG(flag_is_connected);
+
+	err = bt_conn_set_security(g_conn, BT_SECURITY_L2);
+	if (err) {
+		FAIL("Starting encryption procedure failed (%d)\n", err);
+	}
+
+	WAIT_FOR_FLAG(flag_is_encrypted);
+
+	while (bt_eatt_count(g_conn) < CONFIG_BT_EATT_MAX) {
+		k_sleep(K_MSEC(10));
+	}
```

Comment:
Please remind me: Do we establish EATT automatically when the connection is encrypted?